src/components/core/tabs/tabs.tsx

```diff
@@ -1,70 +1,176 @@
-import { Children, ReactElement, useEffect, useRef, useState } from 'react';
-import type { HtmlHTMLAttributes } from 'react';
+import { HtmlHTMLAttributes, ReactNode } from 'react';
+import * as TabsPrimitive from '@radix-ui/react-tabs';
+import useTheme from '@core/theme-context/use-theme';
 import * as Stitches from '@stitches/react';
-import { modifyVariantsForStory } from 'Styles/type-utils';
 import { styled } from 'Styles/stitches.config';
-import type { TabProps } from './tab';
-import Tab from './tab';
+import { modifyVariantsForStory } from 'Styles/type-utils';
+
 export interface TabsProps extends HtmlHTMLAttributes<HTMLDivElement> {
-    active_index?: number;
     contained?: boolean;
     size?: 'default' | 'small';
-    children?: ReactElement<TabProps> | ReactElement<TabProps>[];
+    default_selected: string;
+    tabs: {
+        label: string;
+        icon?: string;
+        iconAlt?: string;
+        content: ReactNode | ReactNode[];
+    }[];
```

Comment:
Can we make the tabs declarative just link `radix-ui`?
We should follow the same declarative pattern that we followed for Accordion.
<img width="601" alt="image" src="https://user-images.githubusercontent.com/56330681/186339163-6bbe5d20-8cbc-4384-9b51-e1c916e15cb1.png">